SimG4Core/Application/interface/ExceptionHandler.h

```diff
@@ -30,8 +30,8 @@ class ExceptionHandler : public G4VExceptionHandler {
               const char *description) override;
 
 private:
-  ExceptionHandler(const ExceptionHandler &) = delete;
-  ExceptionHandler &operator=(const ExceptionHandler &right) = delete;
+  ExceptionHandler(const ExceptionHandler &);
```

Comment:
@fabiocos , this removes compilation warning.

Reply:
@civanch for my understanding which one? We do not have warnings in the regular IB, and I do not see them even with gcc9

Reply:
@fabiocos , depend on compiler options - warning is seen with "scram b code-format" and "scram b code-checks"

Reply:
ok,

```
kipping /build/fabiocos/110X/27706/CMSSW_11_0_X_2019-08-07-2300/src/SimG4Core/Application/interface/ExceptionHandler.h. Compile command not found.
clang-tidy -export-fixes tmp/slc7_amd64_gcc700/code-checks/SimG4Core/Application/src/ExceptionHandler.cc.yaml -header-filter '/build/fabiocos/110X/27706/CMSSW_11_0_X_2019-08-07-2300/src/.*' src/SimG4Core/Application/src/ExceptionHandler.cc
329 warnings generated.
warning: argument unused during compilation: '--param vect-max-version-for-alias-checks=50' [clang-diagnostic-unused-command-line-argument]
/build/fabiocos/110X/27706/CMSSW_11_0_X_2019-08-07-2300/src/SimG4Core/Application/interface/ExceptionHandler.h:34:3: warning: deleted member function should be public [modernize-use-equals-delete]
  ExceptionHandler(const ExceptionHandler &) = delete;
  ^
/build/fabiocos/110X/27706/CMSSW_11_0_X_2019-08-07-2300/src/SimG4Core/Application/interface/ExceptionHandler.h:35:21: warning: deleted member function should be public [modernize-use-equals-delete]
  ExceptionHandler &operator=(const ExceptionHandler &right) = delete;
                    ^
Suppressed 326 warnings (325 in non-user code, 1 with check filters).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
```

Reply:
@civanch I understand that the purpose of the ```= delete``` declaration was to prevent the use of that function, so it could still work if moved to public. On the other hand the present form should practically obtain the same effect (no other class can call these functions), so I would consider mostly a matter of taste than functionality.

Reply:
In my opinion we should follow the modern `= delete` pattern.

Reply:
@makortel for my understanding, do you see any functional difference or is it mostly a matter of old-fashioned/modern code style? From https://en.cppreference.com/w/cpp/language/default_constructor and https://abseil.io/tips/143 I would say the latter, but I would like to know your opinion

Reply:
Defining a function `= delete` explicitly denotes the purpose (a function should not be generated) and is thus self-documenting, whereas declaring a function and leaving out the definition (whether public or private) is a(n old fashioned) mechanism to implement "a deleted function".

While they both achieve the same (you can't call the function), the former is more clear (one doesn't have to wonder whether the definition is left out on purpose or by mistake) and leads to clearer error messages (compiler saying "you're trying to call a deleted function" vs. either compiler saying "you're trying to call a private function" or linker saying "function definition does not exist").

Reply:
@makortel ok, thanks, we do not expect practical difference but it is preferable. I see that @civanch has adopted the suggestion already